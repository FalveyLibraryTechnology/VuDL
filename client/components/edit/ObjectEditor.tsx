import React from "react";
import PropTypes from "prop-types";
import ChildList from "./ChildList";
import Link from "next/link";

const ObjectEditor = ({ pid }) => {
    return (
        <div>
            <h1>Editor: Object {pid}</h1>
            <h2>Tools</h2>
            <ul>
                <li>
                    <Link href={`/edit/object/${pid}/newChild`}>Create New Child Object</Link>
                </li>
            </ul>
            <h2>Contents</h2>
            <ChildList pid={pid} />
        </div>
    );
};

ObjectEditor.propTypes = {
    pid: PropTypes.string,
};

export default ObjectEditor;
