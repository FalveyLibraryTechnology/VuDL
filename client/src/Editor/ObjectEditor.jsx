import React from "react";
import PropTypes from "prop-types";

import AjaxHelper from "../AjaxHelper";
import ChildList from "./ChildList";
import MetadataEditor from "./MetadataEditor";
import Modal from "./Modal";

import "../css/editor.css";

class ObjectEditor extends React.Component {
    static propTypes = {
        pid: PropTypes.string,
    };

    constructor(props) {
        super(props);
        this.ajax = AjaxHelper.getInstance();
        this.state = { modal: null };
        // TODO: Get details
        this.openMetadataEditor = this.openMetadataEditor.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }

    openMetadataEditor(pid) {
        this.setState({
            modal: <MetadataEditor pid={pid} />,
            modalTitle: "Metadata for " + pid,
        });
        this.render();
    }

    closeModal() {
        this.setState({ modal: null });
    }

    render() {
        return (
            <div>
                <h1>{this.props.pid}</h1>
                <ChildList parentID={this.props.pid} openMetadataEditor={this.openMetadataEditor} />

                <Modal title={this.state.modalTitle ?? ""} open={this.state.modal != null} closeModal={this.closeModal}>
                    {this.state.modal ?? ""}
                </Modal>
            </div>
        );
    }
}

export default ObjectEditor;
