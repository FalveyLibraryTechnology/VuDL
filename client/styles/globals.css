:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
