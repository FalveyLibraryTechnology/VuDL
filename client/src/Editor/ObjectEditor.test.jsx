import React from "react";
import { describe, expect, it } from "@jest/globals";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ObjectEditor from "./ObjectEditor";

describe("ObjectEditor", () => {
    it("renders", () => {
        const wrapper = shallow(<ObjectEditor pid="foo:123" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
