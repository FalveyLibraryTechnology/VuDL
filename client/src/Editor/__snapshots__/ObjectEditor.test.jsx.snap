// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectEditor renders 1`] = `
<div>
  <p>
    You are editing 
    foo:123
  </p>
  <ChildList
    pid="foo:123"
  />
</div>
`;
